editor/editor_native_shader_source_visualizer.cpp

```diff
@@ -30,7 +30,43 @@
 
 #include "editor_native_shader_source_visualizer.h"
 
+#include "editor/code_editor.h"
+#include "editor/editor_settings.h"
+#include "editor/themes/editor_scale.h"
 #include "scene/gui/text_edit.h"
+#include "servers/rendering/shader_language.h"
+
+void EditorNativeShaderSourceVisualizer::_load_theme_settings() {
+	syntax_highlighter->set_number_color(EDITOR_GET("text_editor/theme/highlighting/number_color"));
+	syntax_highlighter->set_symbol_color(EDITOR_GET("text_editor/theme/highlighting/symbol_color"));
+	syntax_highlighter->set_function_color(EDITOR_GET("text_editor/theme/highlighting/function_color"));
+	syntax_highlighter->set_member_variable_color(EDITOR_GET("text_editor/theme/highlighting/member_variable_color"));
+
+	syntax_highlighter->clear_keyword_colors();
+
+	List<String> keywords;
+	ShaderLanguage::get_keyword_list(&keywords);
+	const Color keyword_color = EDITOR_GET("text_editor/theme/highlighting/keyword_color");
+	const Color control_flow_keyword_color = EDITOR_GET("text_editor/theme/highlighting/control_flow_keyword_color");
+
+	for (const String &keyword : keywords) {
+		if (ShaderLanguage::is_control_flow_keyword(keyword)) {
+			syntax_highlighter->add_keyword_color(keyword, control_flow_keyword_color);
+		} else {
+			syntax_highlighter->add_keyword_color(keyword, keyword_color);
+		}
+	}
+
+	// Colorize comments.
+	const Color comment_color = EDITOR_GET("text_editor/theme/highlighting/comment_color");
+	syntax_highlighter->clear_color_regions();
+	syntax_highlighter->add_color_region("/*", "*/", comment_color, false);
+	syntax_highlighter->add_color_region("//", "", comment_color, true);
+
+	// Colorize preprocessor statements.
+	const Color user_type_color = EDITOR_GET("text_editor/theme/highlighting/user_type_color");
+	syntax_highlighter->add_color_region("#", "", user_type_color, true);
```

Comment:
Since we have https://github.com/godotengine/godot/pull/85014 merged, should we add a `syntax_highlighter->set_uint_suffix_enabled()` here too?

Reply:
Sure, feel free to open a PR for this.